Lesson14/dicle/script.js

```diff
@@ -0,0 +1,522 @@
+/*
+3. Lowercase Email
+   - Define a function `normalizeEmail(email)` that returns a lowercased version of the email.
+   - Log the normalized email to the console.
+   - Example: "USER@Example.COM" -> "user@example.com"
+*/
+function normalizeEmail(email) {
+  console.log(email.toLowerCase());
+}
+normalizeEmail('DICLEARPAK@GMAIL.COM');
+/*
+6. File Extension Check
+   - Define a function `checkFileExtension(filename)` that checks if the filename
+     ends with ".pdf" using .endsWith().
+   - If it does, log: "This is a PDF file."
+   - Otherwise, log: "Not a PDF file."
+*/
+function checkFileExtension(filename) {
+  if (filename.endsWith('.pdf')) {
+    console.log('This is a PDF file.');
+  } else {
+    console.log('Not a PDF file.');
+  }
+}
+checkFileExtension('dicle.pdf');
+checkFileExtension('dicle.png');
+
+/*
+7. Compare Numbers (if-else)
+   - Define a function `compareNumbers(a, b)` that:
+     - Logs "a is bigger" if a > b
+     - Logs "b is bigger" if b > a
+     - Logs "Numbers are equal" if they are the same
+*/
+function compareNumbers(a, b) {
+  if (a > b) {
+    console.log('a is bigger');
+  } else if (b > a) {
+    console.log('b is bigger');
+  } else {
+    console.log('Numbers are equal');
+  }
+}
+compareNumbers(5, 7);
+compareNumbers(12, 3);
+compareNumbers(3, 3);
+
+/*
+8. Palindrome Check
+   - Define a function `isPalindrome(str)` that checks if `str` is the same
+     forwards and backwards.
+   - If it is, log: "<str> is a palindrome"
+   - Otherwise, log: "<str> is not a palindrome"
+*/
+function isPalindrome(str) {
+  if (str == str.split('').reverse().join('')) {
+    console.log(str + ' is a palindrome');
+  } else {
+    console.log(str + ' is not a palindrome');
+  }
+}
+isPalindrome('abba');
+isPalindrome('apples');
+/*
+9. String Truncation
+   - Define a function `truncateString(text, maxLength)` that uses slice() to
+     cut the string to `maxLength` characters, then appends "..." if it was too long.
+   - Log the final truncated string.
+*/
+function truncateString(text, maxLength) {
+  if (text.length > maxLength) {
+    console.log((truncated = text.slice(0, maxLength) + '...'));
```

Comment:
It's better not to assign variables inside a console log statement. Also, you don't need that variable.
```suggestion
    console.log(text.slice(0, maxLength) + '...');
```